Monitor - Upgrade pyrsmi to amdsmi python library.

guoshzhao opened the pull request:
**Description**
Upgrade to amdsmi python library since pyrsmi will be retired as AMD guys suggested:

AMD SMI Python Library: https://github.com/ROCm/amdsmi/tree/develop/py-interface
pyrsmi: https://github.com/RadeonOpenCompute/pyrsmi

yukirora commented:
hi @guoshzhao, pls check these error msg from MI300 
![image](https://github.com/microsoft/superbenchmark/assets/37182275/76d48ed7-bdc3-4376-b5a2-ef4fd676bbbb)

guoshzhao commented:
> hi @guoshzhao, pls check these error msg from MI300 ![image](https://private-user-images.githubusercontent.com/37182275/291911388-76d48ed7-bdc3-4376-b5a2-ef4fd676bbbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxMjYwMzksIm5iZiI6MTcwMzEyNTczOSwicGF0aCI6Ii8zNzE4MjI3NS8yOTE5MTEzODgtNzZkNDhlZDctYmRjMy00Mzc2LWI1YTItZWY0ZmQ2NzZiYmJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDAyMjg1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmYTQyN2Q3N2YxN2M1NzcxMjg5Mjc1MzZkODM4ZTEzNTk3M2FjMzcyZjEwNWY5MzYwODdhNWUzMjA3NzYwN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7i7XiL2B6g2_mv878c0DRlBCZMfhmBESFFVmCbdHj34)

Thanks, just checked that GPU utilization and temperature APIs can works on MI250. Looks not supported on MI300.
For GPU memory API, I have fixed it.
For ECC API, the errors are expected, I have changed the log level to 'info'.
Besides, I have change all other log level from 'error' to 'warning' to avoid the misunderstanding when incompitibility happens.

yukirora commented:
can we change the warning to only output once for each benchmark, there's too many warnings in the log by this
